a10_octavia/controller/worker/tasks/glm_tasks.py

```diff
@@ -188,3 +188,32 @@ def execute(self, vthunder):
             LOG.warning("No vthunder therefore license revocation cannot occur.")
             return None
         self.axapi_client.delete.glm_license.post()
+
+
+class ConfigureForwardProxyServer(task.Task):
+
+    @axapi_client_decorator
+    def execute(self, vthunder):
+        if not vthunder:
+            LOG.warning("No vthunder therefore forward proxy server cannot be configured.")
+            return
+        host = CONF.glm_license.proxy_host
+        port = CONF.glm_license.proxy_port
+        username = CONF.glm_license.proxy_username
+        enable_password = CONF.glm_license.proxy_password
+        if host and port:
+            try:
+                if enable_password:
+                    secret_string = CONF.glm_license.proxy_secret_string \
+                        if CONF.glm_license.proxy_secret_string else ""
```

Comment:
Pretty sure this entire block can be collapsed into a single create call wrapped in a try except. We expect that the acos-client will raise an error in the event that the secret string isn't specified. We don't want to continue with creation if the secret string has been specified.

Reply:
I was with the impression that user can use empty password. But checked on vthunder empty password is not supported.